content/ui-patterns/messaging.mdx

```diff
@@ -5,27 +5,159 @@ description: Messaging components are used to provide important and relevant inf
 
-- _Flash alerts are best suited for this type of messaging._
+(image here)
+
+## Message states
+
+<img
+  width="960"
+  alt="A group of labels showing the proper icon and color combination for each message state"
+  src="https://github.com/primer/css/assets/18661030/41d3a706-9aa3-4f05-8bad-5176f49a7d5d"
+/>
+
+Messaging components combine state with message type to provide the appropriate level of prominence. To determine which component to use for your message, start by identifying the message state. From there, follow the decision tree to pick a component best suited for your use case.
+
+### Info
+
+Use the `info` state to highlight messages that help users complete a task or that provide additional context. This state is only used for information that is not critical to the user's experience.
+
+Examples:
+- a setting that impacts the current view
+- a nudge towards a new feature
+- a process is in progress
+- single sign on to access data
+
+<img
+  width="960"
+  alt="draft wip"
+  src="https://github.com/primer/css/assets/18661030/3fa52b6d-7e33-4f27-b0ec-d836375fe21a"
+/>
+
+(image of banner in context)
+(image of callout in context)
+
+### Warning
+
+Use the `warning` state to inform users of a potential issue, an important upcoming change that may impact their experience or a potentially unexpected consequence of an action.
+
+Examples:
+- an account setting will become required soon
+- a degraded experience(link) due to connectivity or GitHub availability
+- a non-destructive action with side-effects like costs or access changes is about to take place
+
+<img
+  width="960"
+  alt="draft wip"
+  src="https://github.com/primer/css/assets/18661030/ae82693a-e18c-425f-9f63-e2077012e76b"
+/>
+
+(image of banner in context)
+(image of callout in context)
+(image of announcement in context)
+
+### Success
+
+Use the `success` state to notify users that the results of an action were successful if it is not apparent from the UI.
+
+Examples:
+- saving an account setting if its otherwise not clear the save has taken place
+- an issue has been transferred successfully
+
+<img
+  width="960"
+  alt="draft wip"
+  src="https://github.com/primer/css/assets/18661030/a3612e09-97b2-4ff5-903a-e85d1587ce50"
+/>
+
+(image of banner in context)
+(image of inline message in context)
+
+### Unavailable
+
+Reserve the `unavailable` state for **inline** system related degraded experiences.
+
+Examples:
+- a row of a data table could not be loaded
+- an item in a list could not be loaded
+
+See [degraded experiences](./degraded-experiences) for more information.
+
+### Critical
+
+Use the `critical` state to notify users that the results of an action were unsuccessful, or to inform that an action is destructive. This state is typically used for form errors and confirmation dialogs.
+
+Examples:
+- form field is invalid upon submitting
+- a delete or transfer button
```

Comment:
Irreversible actions should use a modal dialog. I dont think this is applicable.

Reply:
Yes, but there may be a warning banner within the dialog which would use the state `critical`.

Reply:
Partly yes, but if you have a Dialog with other content 
you don't want to add a confirmation dialog on top.

I think this depends on the situation. I feel like the new deletion dialog is a bit to non-critical and could profit from a banner inside the dialog or at least a critical button

![CleanShot 2024-01-18 at 15 55 39@2x](https://github.com/primer/design/assets/813754/6de1d07b-e38f-41e1-84e5-b2e67c88e0a7)

Reply:
This one is a bit of an odd example because actually clicking that button will show another critical dialog, then a _third_ critical dialog, followed by a request for re-auth.

Reply:
Oh, great. 
> Are you really, really, really sure you want to delete it?

But this does not mean the modal should not have a critical banner mentioning that this will delete your project.

The idea is that it pulls you out of just clicking (which people tend to do when there is a popup with a button). The red banner makes you focus for a sec to make sure you actually know what you are doing.